CodeSniffer/Standards/Generic/Sniffs/PHP/SyntaxSniff.php

```diff
@@ -56,7 +56,7 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
     {
         $phpPath = PHP_CodeSniffer::getConfigData('php_path');
         if ($phpPath === null) {
-            return;
+            $phpPath = PHP_BINARY;
```

Comment:
> PHP 5.3 has been unsupported for over a year, so we just assume PHP_BINARY is available.

The PHP version constraint is set in `composer.json` so it must work.

You can use a combination of above suggestions:

* If constant is defined use it.
* If constant doesn't exist guess the binary ( or maybe it's available via `phpinfo();` somehow )
* If guessing failed work as before: return